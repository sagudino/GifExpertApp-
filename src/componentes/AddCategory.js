import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setinputValue] = useState('');
  const inputChange = (e) => {
    setinputValue(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((cats) => [inputValue, ...cats]);
      setinputValue('');
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <input
        type='text'
        value={inputValue}
        onChange={inputChange}
        placeholder='Busca un gif y pulsta intro...'
      ></input>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
