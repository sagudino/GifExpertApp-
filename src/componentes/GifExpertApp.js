import React, { useState } from 'react';

export const GifExpertApp = () => {
  //const categories = ['Categoría 1', 'Categoría 2', 'Categoría 3'];
  const [categories, setCategories] = useState([
    'Categoría 1',
    'Categoría 2',
    'Categoría 3',
  ]);

  const anadirCategoria = () => {
    setCategories([...categories, 'Categoría 4']);
  };
  return (
    <div className='container'>
      <h1>GifExpertApp</h1>
      <hr />
      <button onClick={anadirCategoria}>Agregar</button>
      <ul>
        {categories.map((cat) => {
          return <li key={cat}>{cat}</li>;
        })}
      </ul>
    </div>
  );
};
