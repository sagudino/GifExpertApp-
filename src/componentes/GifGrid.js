import React, { useState, useEffect } from 'react';
import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);
  // solo se ejecuta cuando el componente se renderiza la primera vez
  useEffect(() => {
    getGifs(category).then((imgs) => {
      setImages(imgs);
    });
  }, [category]);

  //getGifs();
  return (
    <>
      <h3>{category}</h3>
      <div className='card-grid'>
        {images.map((image) => (
          <GifGridItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};
