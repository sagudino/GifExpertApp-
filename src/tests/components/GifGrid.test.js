import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../componentes/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en GifGrid', () => {
  const categoria = 'Categoria prueba';

  // se muestra bien con el snapshot
  test('GifGrid Debe mostrarse corectamente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={categoria} />);
    expect(wrapper).toMatchSnapshot();
  });

  // simulamos que useFetchGifs retornoa infor
  test('Debe mostratrar items cuando se cargan imágenes con useFetchGifs', () => {
    const imgs = [
      {
        id: 'abc',
        url: 'https://wwww.adfasf/image.jpg',
        title: 'titulo',
      },
      {
        id: '123abc',
        url: 'https://wwww.adfasf/image.jpg',
        title: 'titulo',
      },
    ];
    useFetchGifs.mockReturnValue({
      data: imgs,
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={categoria} />);
    expect(wrapper).toMatchSnapshot();
    //que no exista el loading porque ya hay datos
    //expect(wrapper.find('p').exists()).toBe(false);
    //que se muestren todas las imágenes
    expect(wrapper.find('GifGridItem').length).toBe(imgs.length);
  });
});
