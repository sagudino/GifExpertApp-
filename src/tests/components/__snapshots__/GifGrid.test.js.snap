// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid Debe mostratrar items cuando se cargan im√°genes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Categoria prueba
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="titulo"
      url="https://wwww.adfasf/image.jpg"
    />
    <GifGridItem
      id="123abc"
      key="123abc"
      title="titulo"
      url="https://wwww.adfasf/image.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en GifGrid GifGrid Debe mostrarse corectamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Categoria prueba
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
